calendar.rb

```diff
@@ -0,0 +1,98 @@
+require './event.rb'
+class Calendar
+  def initialize
+    @events = {}
+  end  
+
+  def add_event
+    while true
+      print "Enter the name of Event: "
+      name = gets.chomp
+      if !name.empty?
+        break
+      end
+    end
+    print "Enter the details of Event: "
+    detail = gets.chomp
+    print "Enter the venue of Event: "
+    venue  = gets.chomp
+    puts "Enter date"
+    date   = get_date
+    event  = Event.new(name, detail, venue, date)
+    if @events.has_key?(date)
+      @events[date] << event
+    else 
+      event_day = Array.new
+      event_day << event
+      @events.merge!(date => event_day)
+    end
+  end
+  def show_events
+    puts "Enter date: "   
+    date = get_date
+    if @events[date]
+      print "\n"
+      puts "Total number of events on #{date}: #{@events[date].length}"
+      puts "Event Name\t\tEvent Detail\t\tEvent Venue"
+      @events[date].each { |event_info| puts "#{event_info.name}\t\t\t#{event_info.detail}\t\t\t#{event_info.venue}" }
```

Comment:
what if the detail is very long? your formatting will be disturbed. so this is not a good way to display events on a specific date. Display in line breaks. i.e.
Event Name: ------
Event Detail: ------
Event Venue: ------